demos/ota/ota_demo_core_http/ota_demo_core_http.c

```diff
@@ -1649,27 +1619,27 @@ static int32_t connectToS3Server( NetworkContext_t * pNetworkContext,
     xSocketsConfig.recvTimeoutMs = otaexampleHTTPS_TRANSPORT_SEND_RECV_TIMEOUT_MS;
 
     /* Retrieve the address location and length from S3_PRESIGNED_GET_URL. */
-    if( pUrl != NULL )
+    if( pcUrl != NULL )
     {
         /* Retrieve the address location and length from S3_PRESIGNED_GET_URL. */
-        httpStatus = getUrlAddress( pUrl,
-                                    strlen( pUrl ),
-                                    &pAddress,
-                                    &serverHostLength );
+        xHTTPStatus = getUrlAddress( pcUrl,
+                                     strlen( pcUrl ),
+                                     &pcAddress,
+                                     &serverHostLength );
 
-        if( httpStatus != HTTPSuccess )
+        if( xHTTPStatus != HTTPSuccess )
         {
             LogError( ( "URL %s parsing failed. Error code: %d",
-                        pUrl,
-                        httpStatus ) );
+                        pcUrl,
+                        xHTTPStatus ) );
         }
 
         /* serverHost should consist only of the host address. */
-        memcpy( serverHost, pAddress, serverHostLength );
+        memcpy( serverHost, pcAddress, serverHostLength );
         serverHost[ serverHostLength ] = '\0';
     }
 
-    if( returnStatus != EXIT_FAILURE )
+    if( returnStatus != pdFAIL )
```

Comment:
It seems to me that this can never fail. The return status is being set to pdPASS and then is not modified before it gets to this conditional. Should the return status be set to fail if the previous steps fail? Otherwise this step does not make sense.

Reply:
Thanks, updated logic:
https://github.com/aws/amazon-freertos/blob/6723dbfe3e000360c4c165f9e4a53bee38908756/demos/ota/ota_demo_core_http/ota_demo_core_http.c#L1655-L1662